### PR TITLE
Conda-forge windows test issue.

### PR DESCRIPTION
**Pull request by mataton**
This PR should hopefully resolve the issue described in #2088. @qiyunzhu I'm not sure if I'm able to test if this works there, because I don't have maintainer access on the scikit-bio feedstock repo, but it at least passes on windows and linux.

Please complete the following checklist:

* [x] I have read the [contribution guidelines](https://scikit.bio/contribute.html).

* [ ] I have documented all public-facing changes in the [changelog](https://github.com/scikit-bio/scikit-bio/blob/main/CHANGELOG.md).

* [ ] **This pull request includes code, documentation, or other content derived from external source(s).** If this is the case, ensure the external source's license is compatible with scikit-bio's [license](https://github.com/scikit-bio/scikit-bio/blob/main/LICENSE.txt). Include the license in the `licenses` directory and add a comment in the code giving proper attribution. Ensure any other requirements set forth by the license and/or author are satisfied.

  - **It is your responsibility to disclose** code, documentation, or other content derived from external source(s). If you have questions about whether something can be included in the project or how to give proper attribution, include those questions in your pull request and a reviewer will assist you.

* [x] This pull request does not include code, documentation, or other content derived from external source(s).

Note: [This document](https://scikit.bio/devdoc/review.html) may also be helpful to see some of the things code reviewers will be verifying when reviewing your pull request.

### CODE REVIEW

**Comment by qiyunzhu** [comment]
@mataton This is a good attempt, but problem still persists. See [here](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=988750&view=logs&j=171a126d-c574-5c8c-1269-ff3b989e923d&t=623c2ba8-c4f1-582d-6c78-1c7699e2e0a2). My guess is that the code can successfully detect Windows, but some Windows systems have `\n` instead of `\r\n`... Maybe the next attempt would be using [`os.lineseq`](https://docs.python.org/3/library/os.html#os.linesep).

File: skbio/io/tests/test_registry.py [hunk 1]
```diff
@@ -641,7 +640,7 @@ def sniffer(fh):
             return True, {}
 
         with io.open(get_data_path('real_file'), 'r') as fh:
-            pos = 3 if platform.system() == "Windows" else 2
+            pos = 3 if len(os.linesep) == 2 else 2
```
Comment (qiyunzhu) [inline skbio/io/tests/test_registry.py:643]: Might be `pos = len(os.linesep) + 1`?

**Comment by qiyunzhu** [comment]
@mataton Still not working. Error message is the same. See [here](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=990101&view=logs&j=00f5923e-fdef-5026-5091-0d5a0b3d5a2c&t=f42c1d49-d161-5453-1c0e-990f827a8536).

**Comment by mataton** [comment]
@qiyunzhu My hunch is that somehow conda-forge's build is seeking to the 4th byte in the file, when it should only be seeking to the 3rd. I've written a new commit with some code that is just designed to test this theory. Could you run the conda-forge pipeline again and see if we get the same error please?

**Comment by qiyunzhu** [comment]
@mataton I did it and the error persisted... [Link](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=990205&view=logs&jobId=5be07ae1-d8ba-5406-47b6-8e3a3a12f825&j=5be07ae1-d8ba-5406-47b6-8e3a3a12f825&t=c94d5231-acfc-5e67-dafc-77bdd383ead4)

**Pull request closed without merging**